server/notebooks/models.py

```diff
@@ -43,6 +43,7 @@ class NotebookRevision(models.Model):
     title = models.CharField(max_length=Notebook.MAX_TITLE_LENGTH)
     created = models.DateTimeField(auto_now_add=True)
     content = models.TextField(blank=True)
+    is_draft = models.BooleanField(default=True)
```

Comment:
I think perhaps we should not set a default for this field, to force any person using this class to think about what type of revision they're adding. Also, when creating the migration, any currently-existing revisions should *not* be considered draft.

Reply:
@wlach do you mean declare `is_draft` field as `is_draft = models.BooleanField()` and change JavaScript code to pass this parameter explicitly? Also, when creating the migration, Use `False` as a one-off default value for existing rows.

Reply:
> @wlach do you mean declare `is_draft` field as `is_draft = models.BooleanField()` and change JavaScript code to pass this parameter explicitly? 

Yes, to not specifying a default, no to providing an interface to setting it via JavaScript / the web api (at least not yet). You may need to modify the django rest framework code to set `is_draft` is false when creating the actual item, in this file:

https://github.com/iodide-project/iodide/blob/master/server/notebooks/api_views.py

Let me know if you need any help with this.

> Also, when creating the migration, Use `False` as a one-off default value for existing rows.

Yup! :+1:

Reply:
Thanks for the explanation! 

> Let me know if you need any help with this

I am not familiar with Django REST Framework, but I think I can figure it out.